.github/workflows/ci.yml

```diff
@@ -0,0 +1,104 @@
+name: Browser Agent CI
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    container:
+      image: ubuntu:latest
+      options: --user=1001
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+      - name: Install Dependencies
+        run: npm ci
+      - name: Run Linting
+        run: npm run lint
+      - name: Run Formatting Check
+        run: npm run format:check
+
+  unit:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    container:
+      image: ubuntu:latest
+      options: --user=1001
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+      - name: Install Dependencies
+        run: npm ci
+      - name: Run Unit Tests
+        run: npm run unit -- --coverage
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+
+  jil-smoke-chrome:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    container:
+      image: ubuntu:latest
+      options: --user=1001 --network-alias=bam-test-1.nr-local.net
+    defaults:
+      run:
+        shell: bash
+    services:
+      selenium:
+        image: selenium/standalone-chrome
+        ports:
+          - 4444:4444
+        options: --shm-size=4g
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+      - name: Install Dependencies
+        run: npm ci
+      - name: Build Agent
+        run: npm run build:all
+      - name: Run JIL Unit Tests
+        run: node ./tools/jil/bin/cli.js -f merged -b chrome@latest/linux -t 85000 --no-retry --selenium-server=selenium:4444 --unit-only
```

Comment:
`--no-retry` can sometimes be problematic when running the full list of tests

Reply:
That usually only applies to func tests though, so this should be ok since its just the browser tests